time_manager/get_cal_time.F90

```diff
@@ -313,34 +322,50 @@ function get_cal_time(time_increment, units, calendar, permit_calendar_conversio
   increment_days = floor(dt/86400)
   increment_seconds = int(dt - increment_days*86400)
 else
-  call error_mesg('get_cal_time','"'//trim(units)//'" is not an acceptable units attribute of time.'// &
+  call error_mesg('get_calendar_time','"'//trim(units)//'" is not an acceptable units attribute of time.'// &
             & ' It must begin with: "years since", "months since", "days since", "hours since", "minutes since",'// &
             & ' or "seconds since"',FATAL)
 endif
 
 if (calendar_in_i /= calendar_tm_i) then
     if(calendar_in_i == NO_CALENDAR .or. calendar_tm_i == NO_CALENDAR) then
-      call error_mesg('get_cal_time','Cannot do calendar conversion because input calendar is '// &
+      call error_mesg('get_calendar_time','Cannot do calendar conversion because input calendar is '// &
        trim(valid_calendar_types(calendar_in_i))//' and time_manager is using '// &
        trim(valid_calendar_types(calendar_tm_i))// &
        ' Conversion cannot be done if either is NO_CALENDAR',FATAL)
     endif
     call get_date(base_time,year, month, day, hour, minute, second)
-    get_cal_time = set_date(year,month,day,hour,minute,second) + set_time(increment_seconds, increment_days)
-    call get_date(get_cal_time,year,month,day,hour,minute,second)
+    get_calendar_time = set_date(year,month,day,hour,minute,second) + set_time(increment_seconds, increment_days)
+    call get_date(get_calendar_time,year,month,day,hour,minute,second)
     call set_calendar_type(calendar_tm_i)
-    get_cal_time = set_date(year,month,day,hour,minute,second, err_msg=err_msg)
+    get_calendar_time = set_date(year,month,day,hour,minute,second, err_msg=err_msg)
     if(err_msg /= '') then
-      call error_mesg('get_cal_time','Error in function get_cal_time: '//trim(err_msg)// &
+      call error_mesg('get_calendar_time','Error in function get_calendar_time: '//trim(err_msg)// &
                ' Note that the time_manager is using the '//trim(valid_calendar_types(calendar_tm_i))//' calendar '// &
-               'while the calendar type passed to function get_cal_time is '//calendar_in_c,FATAL)
+               'while the calendar type passed to function get_calendar_time is '//calendar_in_c,FATAL)
     endif
 else
-    get_cal_time = base_time + set_time(increment_seconds, increment_days)
+    get_calendar_time = base_time + set_time(increment_seconds, increment_days)
 endif
 
-end function get_cal_time
+end function get_calendar_time
+
 !------------------------------------------------------------------------
+
+!> For mixed precision support, just casts to passed in increment to r8
+function get_calendar_time_wrap(time_increment, units, calendar, permit_calendar_conversion)
```

Comment:
wondering if this function is needed

Reply:
It was needed to compile with current mixed mode so I think it'll be needed

Reply:
do you think using class(*) for time_increment would work?  Or is that going to make things messier..